tests/hps/test_hps_classification.py

```diff
@@ -93,30 +93,38 @@ def test_keys_allowed_in_metadata_but_not_in_config_space(
     classification_config_space: LightingTextClassificationConfigSpace,
 ) -> None:
     df, metadata = classification_hps_run_result
-    assert _flatten(metadata).keys() - classification_config_space.__dict__.keys() == {
+    metadata_keys = {**metadata, **metadata["config"].__dict__}.keys()
+    config_space_keys = classification_config_space.__dict__.keys()
+    assert metadata_keys - config_space_keys == {
+        "eval_batch_size",
         "tokenizer_kwargs",
+        "target_column_name",
+        "model_config_kwargs",
+        "task_train_kwargs",
         "batch_encoding_kwargs",
-        "train_batch_size",
+        "embedding_name_or_path",
+        "load_dataset_kwargs",
         "predict_subset",
+        "task_model_kwargs",
+        "tokenizer_name_or_path",
+        "config",
+        "train_batch_size",
+        "datamodule_kwargs",
+        "early_stopping_kwargs",
         "input_column_name",
-        "load_dataset_kwargs",
-        "target_column_name",
-        "accelerator",
         "dataset_name_or_path",
-        "eval_batch_size",
-        "devices",
-        "embedding_name_or_path",
-        "tokenizer_name_or_path",
+        "dataloader_kwargs",
     }
```

Comment:
Yes, during development it has been changed several times while moving some of the attributes to or from keyword arguments to pipeline args. Pipeline metadata now contains a Config object instead of kwargs that introduce of more level of nesting which the tests must reflect. The kwargs should not be be in config space keys as they are returned by one of the method, namely parse_parameters - config does not stores the kwargs in intself